docs/narr/commandline.rst

```diff
@@ -208,39 +208,53 @@ file.
 
 For example, you want to expose your model to the shell along with the database
 session so that you can mutate the model on an actual database. Here, we'll
-assume your model is stored in the ``myapp.models`` package.
+assume your model is stored in the ``myapp.models`` package and that you're
+using ``pyramid_tm`` to configure a transaction manager on the request as
+``request.tm``.
 
 .. code-block:: ini
    :linenos:
 
    [pshell]
    setup = myapp.lib.pshell.setup
-   m = myapp.models
-   session = myapp.models.DBSession
-   t = transaction
+   models = myapp.models
 
 By defining the ``setup`` callable, we will create the module
 ``myapp.lib.pshell`` containing a callable named ``setup`` that will receive
 the global environment before it is exposed to the shell. Here we mutate the
 environment's request as well as add a new value containing a WebTest version
-of the application to which we can easily submit requests.
+of the application to which we can easily submit requests. The ``setup`` callable can also be a generator which can wrap the entire shell lifecycle, executing code when the shell exits.
 
 .. code-block:: python
     :linenos:
 
     # myapp/lib/pshell.py
+    from contextlib import suppress
+    from transaction.interfaces import NoTransaction
     from webtest import TestApp
 
     def setup(env):
-        env['request'].host = 'www.example.com'
-        env['request'].scheme = 'https'
+        request = env['request']
+        request.host = 'www.example.com'
+        request.scheme = 'https'
+
         env['testapp'] = TestApp(env['app'])
 
-When this INI file is loaded, the extra variables ``m``, ``session`` and ``t``
-will be available for use immediately. Since a ``setup`` callable was also
-specified, it is executed and a new variable ``testapp`` is exposed, and the
-request is configured to generate urls from the host
-``http://www.example.com``. For example:
+        # start a transaction which can be used in the shell
+        request.tm.begin()
+
+        # if using the alchemy cookiecutter, the dbsession is connected
+        # to the transaction manager above
+        env['tm'] = request.tm
+        env['dbsession'] = request.dbsession
+        try:
+            yield
+
+        finally:
+            with suppress(NoTransaction):
+                request.tm.abort()
+
+When this INI file is loaded, the extra variable ``models`` will be available for use immediately. Since a ``setup`` callable was also specified, it is executed and new variables ``testapp``, ``tm`` and ``dbsession`` are exposed, and the request is configured to generate urls from the host ``http://www.example.com``. For example:
```

Comment:
This file has more instances of:

```
``.ini``
```
than "INI", and I see the former more often throughout the docs. Would it be too much trouble to make them consistent?

Also Oxford comma:

```
``tm``, and ``dbsession``
```

and "URLs".

Reply:
fixed